### PR TITLE
Add Cloudflare WAF Rule Support

### PR DESCRIPTION
**Pull request by diamonwiggins**
* Adding a CRD for Cloudflare WAF Rules
* Updated `GetCloudflareAPI` to expect an API Token instead of an API Key

### CODE REVIEW

File: docs/docs/api/wafrules.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# WAF Rule
+
+A `kind: WebApplicationFirewallRule` resource will manage one or more WAF rules in a managed [zone](../zone).
+Each resource can contain a single WAF rule when specified under `spec.rule` vs `spec.rules`.
```
Comment (laverya) [inline docs/docs/api/wafrules.md:4]: why not just ask people to add a single entry in a list

Comment (diamonwiggins) [reply docs/docs/api/wafrules.md:4]: yeah, that does make more sense i think

File: pkg/apis/crds/v1alpha1/webapplicationfirewallrule_types.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	ID        string `json:"id"`
+	PackageID string `json:"packageid,omitempty"`
+	Mode      string `json:"mode"`
+}
+
+// WebApplicationFirewallRuleSpec defines the desired state of WebApplicationFirewallRule
+type WebApplicationFirewallRuleSpec struct {
+	Zone  string     `json:"zone"`
+	Rule  *WAFRule   `json:"rule,omitempty"`
+	Rules []*WAFRule `json:"rules,omitempty"`
```
Comment (laverya) [inline pkg/apis/crds/v1alpha1/webapplicationfirewallrule_types.go:33]: having 'rule' and 'rules' feels like it'll bite us, as I said above

File: pkg/controller/webapplicationfirewallrule/webapplicationfirewallrule.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+					existingRule.Mode = desiredRule.Mode
+				}
+
+				if isChanged {
+					rulesToUpdate = append(rulesToUpdate, existingRule)
+				}
+			}
+		}
+		if !found {
+			// TODO
```
Comment (laverya) [inline pkg/controller/webapplicationfirewallrule/webapplicationfirewallrule.go:67]: this is presumably the spot for the 'delete the rule' logic?

do we have some way to only manage rules with a given set of tags, etc?

**Review by laverya — APPROVED** [review]

File: pkg/controller/webapplicationfirewallrule/webapplicationfirewallrule.go (hunk 3 above)
Comment (diamonwiggins) [reply pkg/controller/webapplicationfirewallrule/webapplicationfirewallrule.go:67]: We don't, but I actually think this section doesn't make sense at all as kubeflare doesn't create any of these rules but only modifies them. They're managed by Cloudflare, so they can't be deleted.

**Pull request merged**